.github/workflows/test.yml

```diff
@@ -23,7 +23,7 @@ jobs:
       run: mkdir -p build
 
     - name: Configure CMake
-      run: cmake -B build -S source -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DPTUNIT=ON
+      run: cmake -B build -S source -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DPTUNIT=ON
```

Comment:
Should this patch be squashed with its parent?

Reply:
I have split it to make it more obvious in review how build matrices are done with github actions :)